packages/docusaurus-theme-search-algolia/src/pages/search/index.js

```diff
@@ -223,6 +224,14 @@ function Search() {
     makeSearch(searchResultState.lastPage);
   }, [searchResultState.lastPage]);
 
+  const qsSearch = new URLSearchParams(useLocation().search).get('q');
+
+  useEffect(() => {
+    if (qsSearch && qsSearch !== searchQuery) {
+      setSearchQuery(qsSearch);
+    }
+  }, [qsSearch]);
```

Comment:
Hey @francoischalifour 

Actually just saw that this QS parsing is already provided by the hook: `  const {searchValue, updateSearchPath} = useSearchQuery();
`